### PR TITLE
Fix errorcodes for bad config block request

### PR DESCRIPTION
**Pull request by kevinhartman**
Updated GetConfigInfoBlock with actual error codes. Verified with hwtests.

### CODE REVIEW

**Comment by Subv** [comment]
Please upload the tests to the hwtests repo.  LGTM

File: src/core/hle/result.h [hunk 1]
```diff
@@ -166,6 +166,8 @@ enum class ErrorLevel : u32 {
     Success = 0,
     Info = 1,
 
+    Unknown1 = 11, // returned by cfg when a block is not found or the is wrong size
```
Comment (linkmauve) [inline src/core/hle/result.h]: “… or has a wrong size”

**Comment by kevinhartman** [comment]
PR for hwtests submitted. Also, I'd messed something up which I realized after running the hwtests with our generated ReturnCodes, so needless to say it was a good idea to make them.

**Comment by LittlePippy** [comment]
hi

On Fri, Feb 20, 2015 at 9:50 PM, Kevin Hartman <notifications@github.com>
wrote:

> PR for hwtests submitted. Also, I'd messed something up which I realized
> after running the hwtests with our generated ReturnCodes, so needless to
> say it was a good idea to make them.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/citra-emu/citra/pull/589#issuecomment-75357511>.
>



-- 
I TOLD MY BOSS THAT I DON'T HAVE A CUP ABOUT ME IN THE GAME WHILE EVERYONE
ELSE HAD A CUP ABOUT THEM IN THE HIT RACING GAME MARIO MART AND ALL I GOT
WAS A DRINKING CUP WITH MY FACE ON IT.

File: src/core/hle/service/cfg/cfg.cpp [hunk 2]
```diff
@@ -43,13 +43,18 @@ ResultCode GetConfigInfoBlock(u32 block_id, u32 size, u32 flag, u8* output) {
     SaveFileConfig* config = reinterpret_cast<SaveFileConfig*>(cfg_config_file_buffer.data());
 
     auto itr = std::find_if(std::begin(config->block_entries), std::end(config->block_entries),
-            [&](const SaveConfigBlockEntry& entry) {
-                return entry.block_id == block_id && entry.size == size && (entry.flags & flag);
-            });
+        [&](const SaveConfigBlockEntry& entry) {
+            return entry.block_id == block_id && (entry.flags & flag);
+        });
```
Comment (yuriks) [inline src/core/hle/service/cfg/cfg.cpp]: Can you avoid unnecessarily re-formatting things? It obscures the actual changes in the diff.

Comment (archshift) [inline src/core/hle/service/cfg/cfg.cpp]: Append `?ws=0` to the end of the diff and it all magically goes away :)

Comment (yuriks) [inline src/core/hle/service/cfg/cfg.cpp]: Not for me it doesn't. :) But that's beside the point anyway.

**Comment by kevinhartman** [comment]
It's hardly a "re-formatting". And besides, in the GitHub diff you should see darker areas of the red and green highlights, which should make it clear that I removed one thing from the predicate.

File: src/core/hle/service/cfg/cfg.cpp (hunk 2 above)
Comment (kevinhartman) [inline src/core/hle/service/cfg/cfg.cpp]: In the GitHub diff you should see darker areas of the red and green highlights, which should make it clear that I removed one thing from the predicate.

Comment (archshift) [inline src/core/hle/service/cfg/cfg.cpp]: Er, sorry, meant `?w=0` :P

**Comment by archshift** [comment]
LGTM. Is there anything in specific holding this up?

**Comment by kevinhartman** [comment]
@archshift I think people are hung up on the related hwtests PR. My opinion is that this PR is good to go. It's a fairly minimal change, the hwtests verify the behavior, and the enumeration values that comprise the error code make sense.

**Comment by archshift** [comment]
If any tests confirm a behavior, that should be enough to allow the merging of a PR that implements that behavior. The tests needn't be committed to the hwtests repo in order for a PR to be accepted, IMO. 

@bunnei and @neobrain, is this good to go?

**Comment by Subv** [comment]
Agreed, IMO it is good as it is. LGTM

**Comment by bunnei** [comment]
LGTM

**Pull request merged**